src/SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests/Features/RegisterRenewConfirm.cs

```diff
@@ -0,0 +1,70 @@
+﻿using AutoFixture;
+using FluentAssertions;
+using NUnit.Framework;
+using SFA.DAS.ApprenticeCommitments.Api.Controllers;
+using SFA.DAS.ApprenticeCommitments.Application.Commands.ChangeApprenticeshipCommand;
+using SFA.DAS.ApprenticeCommitments.Application.Commands.CreateRegistrationCommand;
+using SFA.DAS.ApprenticeCommitments.Application.Commands.VerifyRegistrationCommand;
+using SFA.DAS.ApprenticeCommitments.DTOs;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Mail;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeCommitments.Api.AcceptanceTests.Features
+{
+    public class RegisterRenewConfirm
+    {
+        private Fixture fixture;
+
+        [SetUp]
+        public void Setup()
+        {
+            fixture = new Fixture();
+        }
+
+        [Test]
+        public async Task Test()
+        {
+            var factory = Bindings.Api.CreateApiFactory();
+            var client = factory.CreateClient();
+            var db = Bindings.Database.CreateDbContext();
+
+            var create = fixture.Build<CreateRegistrationCommand>()
+                .With(p => p.Email, (MailAddress adr) => adr.ToString())
+                .Create();
+
+            var r1 = await client.PostValueAsync("apprenticeships", create);
+            r1.EnsureSuccessStatusCode();
+
+            var verify = fixture.Build<VerifyRegistrationCommand>()
+                .With(x => x.ApprenticeId, create.ApprenticeId)
+                .With(p => p.Email, create.Email)
+                .Create();
+
+            var r2 = await client.PostValueAsync("registrations", verify);
+            r2.EnsureSuccessStatusCode();
+
+            var (r3, apprenticeships) = await client.GetValueAsync<List<ApprenticeshipDto>>($"apprentices/{create.ApprenticeId}/apprenticeships");
+            var apprenticeshipId = apprenticeships[0].Id;
+
+            var r4 = await client.PostValueAsync(
+                $"apprentices/{create.ApprenticeId}/apprenticeships/{apprenticeshipId}/EmployerConfirmation",
+                new ConfirmEmployerRequest { EmployerCorrect = true });
+            r4.EnsureSuccessStatusCode();
+
+            var change = fixture.Build<ChangeApprenticeshipCommand>()
+                .With(x => x.ApprenticeshipId, create.ApprenticeshipId)
+                .With(p => p.Email, (MailAddress adr) => adr.ToString())
+                .With(p => p.ApprovedOn, (int days) => create.ApprovedOn.AddDays(days))
+                .Create();
+
+            var r5 = await client.PostValueAsync("apprenticeships/change", change);
+            r5.EnsureSuccessStatusCode();
+
+            (r3, apprenticeships) = await client.GetValueAsync<List<ApprenticeshipDto>>($"apprentices/{create.ApprenticeId}/apprenticeships");
+            r3.Should().Be(HttpStatusCode.OK);
+            apprenticeships.Should().HaveCount(2);
+        }
+    }
+}
```

Comment:
This is in the "Acceptance" tests project so I'm not very worried about testing one unit - whatever that is :)

I think this is a useful adjunct to our other, existing tests.  My goal for this particular test was that I could drive the expected behaviour of the API solely from the API endpoints - just as the APIM will do in real life.

I have added an assertion for the ID and course name.  I don't want to test every property as there are other (unit!) tests that do that and it both detract from the focus of this test and also make it more brittle.

Do you think there is harm in having this extra level of testing in the project?

Reply:
Not too worried about this, but I don't think of this as an Acceptance Test project (i think we only called it that because of specflow), it's just I had experience of applying this pattern once before and as the domain became more complex, the tests became troublesome, but there are definitely benefits. 

No strong objections